src/test/java/com/google/devtools/build/lib/skylark/SkylarkRuleImplementationFunctionsTest.java

```diff
@@ -737,6 +738,24 @@ public void testResolveCommandScript() throws Exception {
     assertMatches("argv[1]", "^.*/resolve_me[.][a-z0-9]+[.]script[.]sh$", argv.get(1));
   }
 
+  @Test
+  public void testResolveTools() throws Exception {
+    evalRuleContextCode(
+        createRuleContext("//foo:resolve_me"),
+        "inputs, input_manifests = ruleContext.resolve_tools(",
```

Comment:
Can we verify these can be passed to ctx.action.run ?

Reply:
Done.